### ISSUE TITLE
trio 0.23.0 - AssertionError in Conda

### ISSUE THREAD

**Issue by anakin87** [python-trio/trio#2855]
### To reproduce

```bash
docker run -it continuumio/miniconda3 /bin/bash

pip install trio

python -c "import trio"
```

**Error**

> Traceback (most recent call last):
  File "/opt/conda/lib/python3.11/site-packages/trio/_subprocess.py", line 50, in <module>
    from os import pidfd_open
ImportError: cannot import name 'pidfd_open' from 'os' (/opt/conda/lib/python3.11/os.py)

> During handling of the above exception, another exception occurred:

> Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/opt/conda/lib/python3.11/site-packages/trio/__init__.py", line 23, in <module>
    from . import abc, from_thread, lowlevel, socket, to_thread
  File "/opt/conda/lib/python3.11/site-packages/trio/lowlevel.py", line 48, in <module>
    from ._subprocess import open_process as open_process
  File "/opt/conda/lib/python3.11/site-packages/trio/_subprocess.py", line 54, in <module>
    assert sys.version_info < (3, 9) or sys.implementation.name != "cpython"
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> AssertionError

### Why

It seems related to https://github.com/python-trio/trio/blob/eef24cb9787aa8f4f1707b4889ccd2a71419e3ce/trio/_subprocess.py#L49-L54

- the `os` package in Conda deviates from std python and does not have `pidfd_open` --> `ImportError` (see a [similar issue](https://github.com/conda/conda/issues/10085#issuecomment-661297467)).
- `sys.platform == "linux"` is `True`
- `sys.version_info < (3, 9) or sys.implementation.name != "cpython"` is `False or False` --> `AssertionError`

**Comment by caner-cetin** [comment]
Having same issue. Bump.

**Comment by caner-cetin** [comment]
0.22.2 also fixes the issue.

**Comment by A5rocks** [comment]
Hi, sorry about this!

Mind saying what linux kernel version you use and whether [`os.pidfd_open`](https://docs.python.org/3/library/os.html#os.pidfd_open) is available?

**Comment by caner-cetin** [comment]
I am not quite sure, but you can check https://hub.docker.com/layers/library/python/3.9-slim-buster/images/sha256-5c6f48d1bb7f592ed9b78b4c4efce099957da5fc0b677b8e79a8ba2bd1b087e8 to see which environment I am running my code.

**Comment by A5rocks** [comment]
FYI fix is released in `trio==0.23.1`
